crates/bevy_ecs/src/query/state.rs

```diff
@@ -366,16 +400,15 @@ impl<Q: WorldQuery, F: WorldQuery> QueryState<Q, F> {
             return Err(QueryEntityError::QueryDoesNotMatch(entity));
         }
         let archetype = &world.archetypes[location.archetype_id];
-        let mut fetch = QF::init(world, &self.fetch_state, last_change_tick, change_tick);
-        let mut filter = <QueryFetch<F> as Fetch>::init(
-            world,
-            &self.filter_state,
-            last_change_tick,
-            change_tick,
-        );
+        // Safety: upheld by function safety invariants
+        let this = self.as_whatever::<NewQ, F>();
```

Comment:
Why is this `F`? I can see that the original code used `F` but I thought that was because `F` was fixed in the old logic but this new logic seems to allow `NewF` to be different.

Reply:
ah yeah I think there are a few redundant `self.as_whatever` around the place where they dont actually do anything because we never do anything but access the `fetch/filter_state` fields which would have the same type as before the fn call